[addons] add addon network functions to new way

AlwinEsch opened the pull request:
This change network related addon to kodi callback functions to new
C++ style.

Related doxygen documentation is already added there.

<!--- Provide a general summary of your change in the Title above -->

## Description
<!--- Describe your change in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch commented:
Note: After https://github.com/xbmc/xbmc/pull/12187 is in have this conflicts (need to do a update then)

MartijnKaijser commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/Network.h:
```diff
@@ -0,0 +1,189 @@
+///
+/// The network module offers functions that allow you to control it.
+///
+/// These are pure static functions them no other initialization need.
```
them?

AlwinEsch replied:
Have removed the string complete, this description here becomes improved during completion C++ addon documentation.

ksooo reviewed (approved):
Code is looking good.

AlwinEsch committed:
[addons] add addon network functions to new way

This change network related addon to kodi callback functions to new
C++ style.

Related doxygen documentation is already added there.

AlwinEsch commented:
jenkins build this please

AlwinEsch commented:
jenkins build this please
